crates/rattler_conda_types/src/match_spec/mod.rs

```diff
@@ -274,6 +338,57 @@ impl MatchSpec {
             channel: spec.channel,
             subdir: spec.subdir,
             namespace: spec.namespace,
+            md5: spec.md5,
+            sha256: spec.sha256,
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::str::FromStr;
+
+    use crate::{
+        package::{IndexJson, PackageFile},
+        MatchSpec, PackageRecord,
+    };
+
+    #[test]
+    fn test_digest_match() {
+        let package_dir = tempfile::tempdir().unwrap();
+        let extract_result = rattler_package_streaming::fs::extract(
+            &crate::get_test_data_dir().join("mamba-1.0.0-py38hecfeebb_2.tar.bz2"),
+            package_dir.path(),
+        )
+        .unwrap();
+
+        let index = IndexJson::from_package_directory(package_dir.path()).unwrap();
+        let package_record = PackageRecord::from_index_json(
+            index,
+            None,
+            Some(extract_result.sha256),
+            Some(extract_result.md5),
+        )
+        .unwrap();
```

Comment:
Instead of completely extracting a package and reading the `index.json` to get a single `PackageRecord` I would prefer to construct a fake `PackageRecord` in place. This speeds up the test considerably.